### ISSUE TITLE
mongoose _id &quotes added

### ISSUE THREAD

**Issue by uptownhr** [pugjs/pug#2369]
I'm not sure what and when this changed but after upgrading from jade to pug, I'm seeing that all the ._id I've used in my jade file are now including &quote around the id string. I'm reading that this is because stringify is called on objects, http://stackoverflow.com/questions/29409265/jade-adding-quote-to-data-variable. How come this was not the case before and is there a way to go back to the functionality it was before? Some kind of setting? Or a specific version number this changed so I can revert. 

Thanks!

**Comment by TimothyGu** [comment]
Can you give a specific example?

See #2305 for a list of breaking changes in Pug.

**Comment by ForbesLindesay** [comment]
The problem is that you can do things like:

```js
('foo' + {toString() { return 'bar'; }, toJSON() { return 'bar'; }})
```

which results in `'foobar'` in JavaScript.  We would see `{toString() { return 'bar'; }, toJSON() { return 'bar'; }}` as an object (correctly) and so we would call `JSON.stringify`, but `JSON.stringify` sees the `toJSON` so it returns `'"bar"'` (i.e. the quoted string `'bar'`).  I think we should probably detect the `toJSON` and call it ourselves to see if that returns a string, and it it does we should not JSON stringify it.

**Comment by dimitarkolev** [comment]
Do you have any estimation when the custom toJSON will be published to npm?

**Comment by TimothyGu** [comment]
Fixed in pug-runtime 2.0.2 (published 11 days ago).
